enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/action/presencecheck/MoveToPresenceCheckVerificationPendingAction.java

```diff
@@ -0,0 +1,66 @@
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.app.onboardingserver.statemachine.action.presencecheck;
+
+import com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationPhase;
+import com.wultra.app.enrollmentserver.model.enumeration.IdentityVerificationStatus;
+import com.wultra.app.enrollmentserver.model.integration.OwnerId;
+import com.wultra.app.onboardingserver.common.database.IdentityVerificationRepository;
+import com.wultra.app.onboardingserver.common.database.entity.IdentityVerificationEntity;
+import com.wultra.app.onboardingserver.statemachine.consts.EventHeaderName;
+import com.wultra.app.onboardingserver.statemachine.consts.ExtendedStateVariable;
+import com.wultra.app.onboardingserver.statemachine.enums.OnboardingEvent;
+import com.wultra.app.onboardingserver.statemachine.enums.OnboardingState;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.statemachine.StateContext;
+import org.springframework.statemachine.action.Action;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Action to move the given identity verification to {@code PRESENCE_CHECK / VERIFICATION_PENDING}.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+@Component
+@Slf4j
+public class MoveToPresenceCheckVerificationPendingAction implements Action<OnboardingState, OnboardingEvent> {
+
+    private final IdentityVerificationRepository identityVerificationRepository;
+
+    @Autowired
+    public MoveToPresenceCheckVerificationPendingAction(final IdentityVerificationRepository identityVerificationRepository) {
+        this.identityVerificationRepository = identityVerificationRepository;
+    }
+
+    @Override
+    @Transactional
+    public void execute(StateContext <OnboardingState, OnboardingEvent> context) {
+        final OwnerId ownerId = (OwnerId) context.getMessageHeader(EventHeaderName.OWNER_ID);
+        final IdentityVerificationEntity identityVerification = context.getExtendedState().get(ExtendedStateVariable.IDENTITY_VERIFICATION, IdentityVerificationEntity.class);
+        moveToDocumentUploadVerificationPending(ownerId, identityVerification);
+    }
+
+    private void moveToDocumentUploadVerificationPending(final OwnerId ownerId, final IdentityVerificationEntity idVerification) {
```

Comment:
I would recommend to have this in a separate service.

Reply:
👍 My first reaction was: why. But I realized that similar patter is in other action classes to, so put it to one place may simplify refactor later. My dream is to change phases elsewhere.